cores/esp32/esp32-hal-uart.c

```diff
@@ -82,6 +84,65 @@ uint32_t _get_effective_baudrate(uint32_t baudrate)
     }
 }
 
+
+void uartOnReceive(uart_t* uart, void(*function)(void))
+{
+    if(uart == NULL || function == NULL) {
+        return;
+    }
+    UART_MUTEX_LOCK();
+    uart->onReceive = function;
+    UART_MUTEX_UNLOCK();
+}
+
+
+static void uart_event_task(void *args)
+{
+    uart_t* uart = (uart_t *)args;
+    uart_event_t event;
+    for(;;) {
+        //Waiting for UART event.
+        if(xQueueReceive(uart->uart_event_queue, (void * )&event, (portTickType)portMAX_DELAY)) {
+            switch(event.type) {
+                //Event of UART receving data
+                case UART_DATA:
+                    if(uart->onReceive) uart->onReceive();
```

Comment:
shouldn't there be a mutex lock here too? what if `onReceive` get's cleared right after it was checked?
TBH I probably would not have added locks to `uartOnReceive` either :) but it is in reality possible to cause an exception.

Reply:
Done!